sdk/monitor/monitor-opentelemetry-exporter/src/utils/spanUtils.ts

```diff
@@ -379,3 +390,79 @@ export function readableSpanToEnvelope(span: ReadableSpan, ikey: string): Envelo
     },
   };
 }
+
+/**
+ * Span Events to Azure envelopes parsing.
+ * @internal
+ */
+export function spanEventsToEnvelopes(span: ReadableSpan, ikey: string): Envelope[] {
+  let envelopes: Envelope[] = [];
+  if (span.events) {
+    span.events.forEach((event: TimedEvent) => {
+      let baseType: "ExceptionData" | "MessageData";
+      const sampleRate = 100;
+      let time = new Date(hrTimeToMilliseconds(event.time));
+      let name = "";
+      let baseData: TelemetryExceptionData | MessageData;
+      const properties = createPropertiesFromSpanAttributes(event.attributes);
```

Comment:
I think you actually want to merge the span attributes (if any) with the event attributes for the envelope.

Reply:
Can you confirm? I was not sure about that and couldn't find details in OpenTelemetry specs, if you can associate the dependency/request to the message/exception you can get all the properties, but not sure if we want to duplicate them here

Reply:
Thats where it gets vender specific, my understanding is that "span" events are considered to be associated with "everything" of the span and therefore should be treated as a "sub-event" of the span (in any visualization).

As we are flattening out the events into envelopes, personally, I would expect that there may be resources and attributes of the span that should be included with the event.

Now the million-dollar question is should we add all of them, or will it be specific to the actual event that is being included... We probably need to find out how Java and Python are or will handle this....

Reply:
Java is mapping span events to "trace" (log) telemetry records based on discussion here: https://github.com/microsoft/common-schema/issues/217#issuecomment-893028597

we don't add span attributes to those trace telemetry records, we only add the event attributes